drivers/openstack/openstack.go

```diff
@@ -454,8 +454,33 @@ func (d *Driver) Kill() error {
 }
 
 func (d *Driver) Remove() error {
+
+	var floatingIP *FloatingIP
+
 	log.Debug("deleting instance...", map[string]string{"MachineId": d.MachineId})
 	log.Info("Deleting OpenStack instance...")
+
+	if err := d.resolveIds(); err != nil {
+		return err
+	}
+	if ips, err := d.client.GetFloatingIPs(d); err != nil {
+		return err
+	} else {
+		for _, xip := range ips {
```

Comment:
As far as I can tell, we should be able to add filters to `GetFloatingIPs` to only retrieve the relevant data and avoid this loop entirely. https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/extensions/layer3/floatingips/requests.go#L8-L13

Reply:
I believe I even tried this but failed to implement it. Do you have a hint on how it should look like in the code when used?

Reply:
Something like this? https://gist.github.com/shin-/84599a9ad3dbc0f0b7872ad259c60c42
You'll want to test this against an actual openstack setup to make sure the filter works as expected, but that's the direction I was thinking of.

Reply:
Thanks, I'll try that

Reply:
Your patch seems to work so far for the filtering.
```
# docker-machine ls
NAME                                                       ACTIVE   DRIVER      STATE     URL                          SWARM   DOCKER        ERRORS
runner-d9685ed3-gitlabci-small-multi-1538471577-2212a5bc   -        openstack   Running   tcp://abc:2376           v18.06.1-ce
runner-d9685ed3-gitlabci-small-multi-1538471580-f9dce75d   -        openstack   Running   tcp://abc2:2376            v18.06.1-ce
runner-d9685ed3-gitlabci-small-multi-1538471583-35298647   -        openstack   Running   tcp://abc3:2376           v18.06.1-ce
```